### ISSUE TITLE
Support JupyterLab 2.0.0

### ISSUE THREAD

**Issue by memeplex** [ryantam626/jupyterlab_code_formatter#108]
```
~:: jupyter labextension install @ryantam626/jupyterlab_code_formatter
An error occured.
ValueError: This extension does not yet support the current version of JupyterLab.


Conflicting Dependencies:
JupyterLab              Extension      Package
>=2.0.0 <2.1.0          >=1.0.0 <2.0.0 @jupyterlab/application
>=2.0.0 <2.1.0          >=1.0.0 <2.0.0 @jupyterlab/fileeditor
>=2.0.0 <2.1.0          >=1.0.0 <2.0.0 @jupyterlab/notebook
```

**Comment by ryantam626** [comment]
Will require some work, there are some breaking changes unfortunately

**Comment by ryantam626** [comment]
Just published version 1.2.0 

- `pip install -U jupyterlab_code_formatter`
- `jupyter labextension install @ryantam626/jupyterlab_code_formatter@v1.2.0`
- Restart jupyterlab

should do the trick.

**Comment by mlucool** [comment]
Are you sure this should not be marked as a breaking change? For example, I don't see LabIcon in [1.2.x](https://github.com/jupyterlab/jupyterlab/tree/1.2.x/packages/ui-components/src/icon) but I do in the [2.x branch](https://github.com/jupyterlab/jupyterlab/tree/master/packages/ui-components/src/icon).

**Comment by ryantam626** [comment]
I knew I forgot something :expressionless: fixing

**Comment by ryantam626** [comment]
> Are you sure this should not be marked as a breaking change? For example, I don't see LabIcon in [1.2.x](https://github.com/jupyterlab/jupyterlab/tree/1.2.x/packages/ui-components/src/icon) but I do in the [2.x branch](https://github.com/jupyterlab/jupyterlab/tree/master/packages/ui-components/src/icon).

What do you mean? https://github.com/ryantam626/jupyterlab_code_formatter/blob/master/labextension/package.json#L36-L39 is already ^2.0.0?

**Comment by michaelaye** [comment]
I am using it for the first time, and in JLab 2.0, are you sure it works? I am trying it with a cell like this:

```python
a='hello'
```
and expect it to be reformatted to:
```python
a = "hello"
```
using black (confirmed in the settings) and black is installed.

I tried Right-click->Format cell, while
* having cell active in editing mode
* having cell selected in command mode
* having the text in the cell selected using the mouse and using the command then.

to no avail.

**Comment by mlucool** [comment]
Did you test this works on 1.x still? I ask because this would start mixing 1.x and 2.x lab dependencies (this brings in 2.x but everything else assumes 1.x). I'm not sure that will work as you hope.

**Comment by ryantam626** [comment]
> Did you test this works on 1.x still? I ask because this would start mixing 1.x and 2.x lab dependencies (this brings in 2.x but everything else assumes 1.x). I'm not sure that will work as you hope.

I am not understanding your question. On 1.x, `jupyter labextension install @ryantam626/jupyterlab_code_formatter` should install for you the v1.1.0 version of this plugin, since there is no change with both the pypi/npm package (I am not sure you can even do this, they only support outright deletion of the versions), why would it not work?

Anyway, I tested with jupyter lab 1.2.5 just now, and npm (or yarn or whatever jupyterlab actually use these days) automatically resolves correctly to v1.1.0 for the labextension, and after manually matching the serverextension version (i.e. `pip install jupyterlab_code_formatter==1.1.0`), it works as expected.

**Comment by ryantam626** [comment]
Seems we are all good for jupyterlab ^2.0.0 now, closing :+1:

**Comment by thomaspernet** [comment]
@ryantam626  It seems the issue is not solved for Jupyterlab 2.0.2. 

I updated all the extensions and jupyterlab_code_formatter isn't showing on the left panel. jupyterlab_code_formatter is available in the labextension and server. I use `autopep8`

`JupyterLab v2.0.1
Known labextensions:
   app dir: /Users/thomas/anaconda3/share/jupyter/lab
        @aquirdturtle/collapsible_headings v2.2.0  enabled  OK
        @jupyter-widgets/jupyterlab-manager v2.0.0  enabled  OK
        @jupyterlab/github v2.0.0 disabled OK
        @jupyterlab/latex v2.0.0  enabled  OK
        @jupyterlab/toc v3.0.0  enabled  OK
        @ryantam626/jupyterlab_code_formatter v1.2.2  enabled  OK
        jupyter-matplotlib v0.7.2  enabled  OK
        jupyterlab-jupytext v1.2.1  enabled  OK
        jupyterlab-sos v0.6.1  enabled  OK
        nbdime-jupyterlab v2.0.0  enabled  OK
        qgrid v1.1.1  enabled  OK
        transient-display-data v0.4.0  enabled  OK`

and 

`config dir: /Users/thomas/.jupyter
    jupyterlab_git  enabled 
    - Validating...
      jupyterlab_git  OK
    jupyterlab_code_formatter  enabled 
    - Validating...
      jupyterlab_code_formatter  OK
    jupyter_nbextensions_configurator  enabled 
    - Validating...
      jupyter_nbextensions_configurator 0.4.1 OK
    hide_code  enabled 
    - Validating...
      hide_code  OK
config dir: /Users/thomas/anaconda3/etc/jupyter
    jupyterlab  enabled 
    - Validating...
      jupyterlab 2.0.1 OK
    jupyterlab_github  enabled 
    - Validating...
      jupyterlab_github 1.0.0 OK
    jupyterlab_latex  enabled 
    - Validating...
      jupyterlab_latex 1.0.0 OK
    jupytext  enabled 
    - Validating...
      jupytext 1.4.0+dev OK
    nbdime  enabled 
    - Validating...
      nbdime 1.1.0 OK`

**Comment by ryantam626** [comment]
hi @thomaspernet please open a new issue and i can follow up there
